### PR TITLE
[AC-7058] Track api calls

### PR DESCRIPTION
**Pull request by kaymation**
[JIRA](https://masschallenge.atlassian.net/browse/AC-7058)

Log api calls to get more insight into what endpoints are being hit, and by whom

TO TEST:
 - Log onto impact api on test5
 - Load various endpoints
- VERIFY: info is being logged at https://papertrailapp.com/systems/impact-api/events

### CODE REVIEW

**Comment by pbx** [comment]
Taking code review.

File: web/impact/impact/settings.py [hunk 1]
```diff
@@ -279,6 +285,80 @@ class Base(Configuration):
...
+                'class': 'logging.handlers.RotatingFileHandler',
+                'maxBytes': 1024 * 1024 * 5,
+                'backupCount': 5,
+                'formatter': 'detail',
+                'filename': 'debug.log',
+            },
+            'mail_admins': {
+                'level': 'ERROR',
+                'filters': ['require_debug_false'],
+                'class': 'django.utils.log.AdminEmailHandler'
```
Comment (pbx) [inline web/impact/impact/settings.py:315]: Adding `mail_admins` handling seems outside the scope of the ticket.

**Review by pbx — CHANGES_REQUESTED** [review]
When I ran the tests locally, there was a lot of noise that is not seen running tests on development -- warnings from `django.request`, and GraphQL tracebacks.
```
$ make test
WARNING: Found orphan containers (mc-mysql) for this project. If you removed or renamed this service in your compose file, you can run this command with the --remove-orphans flag to clean it up.
Starting impact-api_redis_1     ... done
Starting impact-api_front-end_1 ... done
Starting impact-api_assets_1    ... done
Using existing test database for alias 'default'...
System check identified no issues (0 silenced).
........................................................................................host: localhost  django.request[1]: WARNING Not Found: /api/v1/criterion/1/
host: localhost  django.request[1]: WARNING Not Found: /api/v1/criterion/1/
host: localhost  django.request[1]: WARNING Not Found: /api/v1/criterion/1/
...........host: localhost  django.request[1]: WARNING Not Found: /api/v1/criterion_option_spec/1/
host: localhost  django.request[1]: WARNING Not Found: /api/v1/criterion_option_spec/1/
host: localhost  django.request[1]: WARNING Not Found: /api/v1/criterion_option_spec/1/
..................host: localhost  graphql.execution.executor[1]: ERROR An error occurred while resolving field Query.expertProfile
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/graphql/execution/executor.py", line 447, in resolve_or_error
    return executor.execute(resolve_fn, source, info, **args)
  File "/usr/local/lib/python3.6/site-packages/graphql/execution/executors/sync.py", line 16, in execute
    return fn(*args, **kwargs)
  File "/wwwroot/impact/graphql/middleware.py", line 8, in resolve
    raise ResponseError(message=NOT_LOGGED_IN_MSG, code="401")
impact.graphql.utils.response_error.ResponseError: User Not Logged in!
host: localhost  graphql.execution.utils[1]: ERROR Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/graphql/execution/executor.py", line 447, in resolve_or_error
    return executor.execute(resolve_fn, source, info, **args)
  File "/usr/local/lib/python3.6/site-packages/graphql/execution/executors/sync.py", line 16, in execute
    return fn(*args, **kwargs)
  File "/wwwroot/impact/graphql/middleware.py", line 8, in resolve
    raise ResponseError(message=NOT_LOGGED_IN_MSG, code="401")
graphql.error.located_error.GraphQLLocatedError: User Not Logged in!

./usr/local/lib/python3.6/site-packages/django/db/models/fields/__init__.py:1451: RuntimeWarning: DateTimeField Program.end_date received a naive datetime (2019-10-30 10:04:17.424236) while time zone support is active.
  RuntimeWarning)
...host: localhost  graphql.execution.utils[1]: ERROR Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/promise/promise.py", line 85, in try_catch
    return (handler(*args, **kwargs), None)
  File "/usr/local/lib/python3.6/site-packages/graphql/execution/executor.py", line 527, in <lambda>
    exe_context, return_type, field_asts, info, path, resolved
  File "/usr/local/lib/python3.6/site-packages/graphql/execution/executor.py", line 536, in complete_value
    raise GraphQLLocatedError(field_asts, original_error=result, path=path)
graphql.error.located_error.GraphQLLocatedError: Entrepreneur matching the id does not exist.

.host: localhost  graphql.execution.utils[1]: ERROR Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/promise/promise.py", line 85, in try_catch
    return (handler(*args, **kwargs), None)
  File "/usr/local/lib/python3.6/site-packages/graphql/execution/executor.py", line 527, in <lambda>
    exe_context, return_type, field_asts, info, path, resolved
  File "/usr/local/lib/python3.6/site-packages/graphql/execution/executor.py", line 536, in complete_value
    raise GraphQLLocatedError(field_asts, original_error=result, path=path)
graphql.error.located_error.GraphQLLocatedError: Expert matching the id does not exist.

............................................./usr/local/lib/python3.6/site-packages/django/db/models/fields/__init__.py:1451: RuntimeWarning: DateTimeField MentorProgramOfficeHour.start_date_time received a naive datetime (2019-10-30 14:04:23.240032) while time zone support is active.
  RuntimeWarning)
...host: localhost  django.request[1]: WARNING Not Found: /api/v0/mentors/
host: localhost  django.request[1]: WARNING Not Found: /api/v0/mentors/
host: localhost  django.request[1]: WARNING Not Found: /api/v0/mentors/
.host: localhost  django.request[1]: WARNING Not Found: /api/v0/mentors/
host: localhost  django.request[1]: WARNING Not Found: /api/v0/mentors/
host: localhost  django.request[1]: WARNING Not Found: /api/v0/mentors/
......host: localhost  django.request[1]: WARNING Not Found: /oauth/applications/1/
host: localhost  django.request[1]: WARNING Not Found: /oauth/applications/1/
host: localhost  django.request[1]: WARNING Not Found: /oauth/applications/1/
...................................................................................host: localhost  django.request[1]: WARNING Not Found: /api/v0/startup_detail/
host: localhost  django.request[1]: WARNING Not Found: /api/v0/startup_detail/
host: localhost  django.request[1]: WARNING Not Found: /api/v0/startup_detail/
..host: localhost  django.request[1]: WARNING Not Found: /api/v0/startup_detail/
host: localhost  django.request[1]: WARNING Not Found: /api/v0/startup_detail/
host: localhost  django.request[1]: WARNING Not Found: /api/v0/startup_detail/
.........host: localhost  django.request[1]: WARNING Not Found: /api/v0/startup_list/
host: localhost  django.request[1]: WARNING Not Found: /api/v0/startup_list/
host: localhost  django.request[1]: WARNING Not Found: /api/v0/startup_list/
.host: localhost  django.request[1]: WARNING Not Found: /api/v0/startup_list/
host: localhost  django.request[1]: WARNING Not Found: /api/v0/startup_list/
host: localhost  django.request[1]: WARNING Not Found: /api/v0/startup_list/
...................................host: localhost  django.request[1]: WARNING Not Found: /api/v1/user/2/
host: localhost  django.request[1]: WARNING Not Found: /api/v1/user/2/
host: localhost  django.request[1]: WARNING Not Found: /api/v1/user/2/
...................................................................................
----------------------------------------------------------------------
Ran 390 tests in 40.244s

OK
```

File: web/impact/impact/settings.py (hunk 1 above)
Comment (kaymation) [reply web/impact/impact/settings.py:315]: the `LOGGING` parameter must be defined with all the expected keys, including `mail_admins`, or we face errros

Comment (pbx) [reply web/impact/impact/settings.py:315]: Interesting. If we agree that we don't mind adding `mail_admins` here (and we figure out in advance of deployment roughly what we expect the message volume to be), I can live with adding it.

Otherwise, re disabling it, a couple possible leads I found:
 * [disabling the automatic logging config](https://docs.djangoproject.com/en/1.11/topics/logging/#disabling-logging-configuration)
* using `logging.NullHandler` [(example)](https://docs.djangoproject.com/en/1.11/topics/logging/#django-security)
* a few [ideas from Lincoln Loop](https://lincolnloop.com/blog/disabling-error-emails-django/)

**Comment by pbx** [comment]
@kaymation - local test run is clean now, thanks. What is the URL for impact-api in the test5 stack?

_Answer: https://test5api.masschallenge.org/api/accelerator/_

**Review by pbx — APPROVED** [review]
Calls are indeed being logged to Papertrail. Thanks for your tenacity.

**Review by philophilo — CHANGES_REQUESTED** [review]
Just a few linting issues
```
web/impact/impact/settings.py:5:1: F401 'sys' imported but unused
web/impact/impact/settings.py:6:1: F401 'logging' imported but unused
web/impact/impact/settings.py:395:1: W293 blank line contains whitespace
```

**Review by philophilo — APPROVED** [review]
Looks good
